mapie/tests/test_calibration.py

```diff
@@ -0,0 +1,390 @@
+    calibrator: Union[str, RegressorMixin],
+    estimator: ClassifierMixin
+) -> None:
+    """Test that the size of the outputs are coherent."""
+    mapie_cal = MapieCalibrator(
+        estimator=estimator,
+        calibrator=calibrator,
+    )
+    mapie_cal.fit(X, y)
+    calib_ = mapie_cal.predict_proba(X)
+    assert calib_.shape == (len(y), mapie_cal.n_classes_)
+
+
+def test_number_of_classes_equal_calibrators() -> None:
+    """
+    Test that the number of calibrators is the same as the number
+    of classes in the calibration step.
+    """
+    mapie_cal = MapieCalibrator()
+    mapie_cal.fit(
+        X=X_,
+        y=y_,
+        random_state=random_state
+    )
+    y_pred_calib_set = mapie_cal.estimator.predict(X=X_calib)  # type: ignore
+    assert len(mapie_cal.calibrators) == len(np.unique(y_pred_calib_set))
+
+
+def test_same_predict() -> None:
+    """Test that the same prediction is made regardless of the calibration."""
+    mapie_cal = MapieCalibrator(method="top_label")
+    mapie_cal.fit(
+        X=X_,
+        y=y_,
+        random_state=random_state
+    )
+    y_pred_calib_set = mapie_cal.estimator.predict(X=X_test)  # type: ignore
+    y_pred_calibrated_test_set = mapie_cal.predict(X=X_test)
+    np.testing.assert_allclose(y_pred_calib_set, y_pred_calibrated_test_set)
+
+
+def test_correct_results() -> None:
+    """
+    Test that the y_score and top label score from the test dataset result
+    in the correct scores (in a multi-class setting).
+    """
+    mapie_cal = MapieCalibrator(cv="split")
+    mapie_cal.fit(
+        X=X_,
+        y=y_,
+        random_state=random_state
+    )
+    pred_ = mapie_cal.predict_proba(X_test)
+    top_label_ece_ = top_label_ece(y_test, pred_)
+    np.testing.assert_array_almost_equal(
+        results["y_score"], pred_  # type:ignore
+    )
+    np.testing.assert_allclose(  # type:ignore
+        results["top_label_ece"],
+        top_label_ece_,
+        rtol=1e-2
+    )
+
+
+def test_correct_results_binary() -> None:
+    """
+    Test that the y_score and top label score from the test dataset result
+    in the correct scores (in a binary setting).
+    """
+    X_binary, y_binary = make_classification(
+        n_samples=10,
+        n_classes=2,
+        n_informative=4,
+        random_state=random_state
+    )
+    mapie_cal = MapieCalibrator()
+    mapie_cal.fit(
+        X=X_binary,
+        y=y_binary,
+        random_state=random_state
+    )
+    pred_ = mapie_cal.predict_proba(X_binary)
+    top_label_ece_ = top_label_ece(y_binary, pred_)
+    ece = expected_calibration_error(y_binary, pred_)
+    np.testing.assert_array_almost_equal(
+        results_binary["y_score"], pred_  # type:ignore
+    )
+    np.testing.assert_allclose(  # type:ignore
+        results_binary["top_label_ece"],
+        top_label_ece_,
+        rtol=1e-2
+    )
+    np.testing.assert_allclose(  # type:ignore
+        results_binary["ece"],
+        ece,
+        rtol=1e-2
+    )
+
```

Comment:
Please add a test with more than 2 classes

Reply:
Not sure what you mean here, let's discuss tomorrow.